.github/workflows/benchmarks.yml

```diff
@@ -143,6 +147,46 @@ jobs:
         with:
           path: asv-compare-changed-output.log
 
+      - name: Run benchmarks for base to head commits of PR
+        if: github.event_name == 'pull_request_target'
+        run: |
+          echo $(git rev-parse HEAD) > commit_hashes.txt
+          echo $(git merge-base HEAD upstream/master) >> commit_hashes.txt
+
+          asv run HASHFILE:commit_hashes.txt | tee asv-output-PR.log  
+            if grep -q failed asv-output-PR.log; then 
+              echo "Some benchmarks have failed!"
+              exit 1
+            fi
```

Comment:
So if a pull request has like 15 commits, it will run on all of them?

Reply:
No, only 2

Reply:
I see